boat-maven-plugin/src/main/java/com/backbase/oss/boat/BoatBayRadio.java

```diff
@@ -0,0 +1,175 @@
+package com.backbase.oss.boat;
+
+import com.backbase.oss.boat.bay.client.ApiClient;
+import com.backbase.oss.boat.bay.client.api.UploadPluginApi;
+import com.backbase.oss.boat.bay.client.model.*;
+import com.backbase.oss.boat.loader.OpenAPILoader;
+import com.backbase.oss.boat.mapper.LintReportMapperImpl;
+import com.fasterxml.jackson.databind.introspect.TypeResolutionContext.Basic;
+import feign.auth.BasicAuthRequestInterceptor;
+import io.swagger.v3.oas.models.OpenAPI;
+import kotlin.ranges.IntRange;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.IOUtils;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+import org.springframework.expression.Expression;
+import org.springframework.expression.ExpressionParser;
+import org.springframework.expression.spel.standard.SpelExpressionParser;
+import org.zalando.zally.rule.api.Severity;
+
+import java.io.File;
+import java.nio.charset.Charset;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Slf4j
+public class BoatBayRadio {
+
+  @Parameter(name = "readTimeout", defaultValue = "null")
+  public Integer readTimeout;
+
+  private final UploadPluginApi boatbayUploadSpecClient;
+  private final File inputSpec;
+  private String clientBasePath;
+  private LintReportMapperImpl reportMapper;
+  protected MavenProject project;
+
+
+  public BoatBayRadio(File inputSpec, MavenProject project, String clientBasePath) {
+    this.project = project;
+    this.reportMapper = new LintReportMapperImpl();
+    this.inputSpec = inputSpec;
+    this.clientBasePath = clientBasePath;
+
+    if (System.getenv().containsKey("BOAT_BAY_SERVER_URL"))
+      this.clientBasePath= System.getenv("BOAT_BAY_SERVER_URL");
+
+    boatbayUploadSpecClient = configureApiClient(this.clientBasePath).buildClient(UploadPluginApi.class);
+  }
+
+  private ApiClient configureApiClient(String clientBasePath){
+
+    if (readTimeout == null){
+      BasicAuthRequestInterceptor basicAuthRequestInterceptor = new BasicAuthRequestInterceptor("admin","admin");
+      ApiClient apiClient = new ApiClient().setBasePath(clientBasePath);
+//      apiClient.addAuthorization("Basic",basicAuthRequestInterceptor);
```

Comment:
How is auth working now?

Reply:
I guess, by default some basic security is enabled by Jhipster. If you spin up boat-bay locally, in the mysql database in jhi_user, you would see 2 users - admin(ROLE_ADMIN) and user (ROLE_USER).
To create something, we need to pass as basic auth.

Reply:
No security applied